lib/rubocop/cop/rails/hash_enum.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop looks for enums written with array syntax.
+      #
+      # When using array syntax, adding an element in a
+      # position other than the last causes all previous
+      # definitions to shift. Explicitly specifying the
+      # value for each key prevents this from happening.
+      #
+      # @example
+      #   # bad
+      #   enum status: [:active, :archived]
+      #
+      #   # good
+      #   enum status: { active: 0, archived: 1 }
+      #
+      class HashEnum < Cop
```

Comment:
I think that `Rails/EnumHash` cop is a better cop name because `Rails/EnumUniqueness` cop already exists.
https://docs.rubocop.org/projects/rails/en/stable/cops_rails/#railsenumuniqueness